exist-core/src/main/java/org/exist/management/client/JMXServlet.java

```diff
@@ -218,32 +214,34 @@ public void init(ServletConfig config) throws ServletException {
      * Register all known IP-addresses for localhost.
      */
     void registerLocalHostAddresses() {
-        // The external IP address of the server
-        try {
-            localhostAddresses.add(InetAddress.getLocalHost().getHostAddress());
-        } catch (UnknownHostException ex) {
-            LOG.warn("Unable to get HostAddress for localhost: {}", ex.getMessage());
-        }
 
-        // The configured Localhost addresses
         try {
-            for (InetAddress address : InetAddress.getAllByName("localhost")) {
-                localhostAddresses.add(address.getHostAddress());
+            for (final NetworkInterface networkInterface : NetworkInterface.networkInterfaces().toList()) {
+                if (networkInterface.isLoopback()) {
+                    final Enumeration<InetAddress> inetAddresses = networkInterface.getInetAddresses();
+                    while (inetAddresses.hasMoreElements()) {
+                        final String hostAddress = inetAddresses.nextElement().getHostAddress();
+                        if (hostAddress != null) {
+                            if (hostAddress.contains("%")) {
+                                final int position = hostAddress.indexOf("%");
+                                localhostAddresses.add(hostAddress.substring(0, position));
+                            } else {
+                                localhostAddresses.add(hostAddress);
+                            }
             }
-        } catch (UnknownHostException ex) {
-            LOG.warn("Unable to retrieve ipaddresses (v4) for localhost: {}", ex.getMessage());
         }
 
-        try {
-            for (InetAddress address : Inet6Address.getAllByName("localhost") ){
-                localhostAddresses.add(address.getHostAddress());
             }
-        } catch (UnknownHostException ex) {
-            LOG.warn("Unable to retrieve ipaddresses (v6) for localhost: {}", ex.getMessage());
+            }
+
+        } catch (final SocketException e) {
+            LOG.error("Unable to determine localhost loopback addresses: {}  ", e.getMessage(), e);
         }
```

Comment:
Could be rewritten using streams and a simple additional method:
```
NetworkInterface.networkInterfaces()
    .filter(SimpleTest::isLoopback)
    .flatMap(NetworkInterface::inetAddresses)
    .map(InetAddress::getHostAddress)
    .map(a -> a.replaceFirst("%.+", ""))
    .forEach(localhostAddresses::add);
```
Additonal Method would be:
```
static boolean isLoopback(NetworkInterface ne) {
    try {
        return ne.isLoopback();
    } catch (SocketException e) {
        LOG.error("Unable to determine if {} is a loopback interface", ne, e);
        return false;
    }
}
```

Reply:
@reinhapa I doubted about this due to readability with the "%" but your suggestion really looks  clean!